### PR TITLE
core: Add BlockChainConfig 

### PR DESCRIPTION
**Pull request by omkar-mohanty**
In reference to  #20280 this PR adds a `core.BlockChainConfig` struct to simplify the `core.NewBlockChain` function.

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -143,6 +143,55 @@ var defaultCacheConfig = &CacheConfig{
 	SnapshotWait:   true,
 }
 
+type BlockChainConfig struct {
```
Comment (theblueskies) [inline core/blockchain.go:146]: Exported structs like this should have comments explaining them

File: core/blockchain.go [hunk 2]
```diff
@@ -143,6 +143,55 @@ var defaultCacheConfig = &CacheConfig{
...
+	TxLookupLimit  *uint64
+}
+type BcConfigOpt func(*BlockChainConfig)
+
+func SetCacheConfig(cacheConfig *CacheConfig) BcConfigOpt {
+	return func(bcConfig *BlockChainConfig) {
+		bcConfig.CacheConfig = cacheConfig
+	}
+}
+func SetChainConfig(chainConfig *params.ChainConfig) BcConfigOpt {
```
Comment (theblueskies) [inline core/blockchain.go:160]: Exported functions like this should have comments explaining them

File: core/blockchain.go (hunk 1 above)
Comment (omkar-mohanty) [reply core/blockchain.go:149]: Thanks for the suggestion I've added comment lines to explain `core.BlockChainConfig` struct.

File: core/blockchain.go (hunk 2 above)
Comment (omkar-mohanty) [reply core/blockchain.go:170]: Yep  thanks for the suggestion and I've added comment lines explaining each of the functions added.

**Comment by holiman** [comment]
Sorry, we were too slow on the ball here (1 year later) , and it's by now heavily bitrotted. IT's on our backlog to look at though

**Comment by fjl** [comment]
We will probably do a refactoring like this at some point, but there is no need to keep this PR open.

**Pull request closed without merging**